web/frontend/src/components/vision/index.svelte

```diff
@@ -0,0 +1,265 @@
+<script lang="ts">
+import Collapse from '@/lib/components/collapse.svelte';
+import { useRobotClient } from '@/hooks/robot-client';
+import { components } from '@/stores/resources';
+import {
+  CameraClient,
+  VisionClient,
+  type Classification,
+  type Detection,
+  type ServiceError,
+} from '@viamrobotics/sdk';
+import { rcLogConditionally } from '../../lib/log';
+import { displayError } from '@/lib/error';
+import { filterSubtype } from '../../lib/resource';
+import get from 'lodash-es/get';
+
+export let names: string[];
+let [serviceName] = names;
+
+$: cameras = filterSubtype($components, 'camera');
+
+const { robotClient } = useRobotClient();
+
+$: visionClient = new VisionClient($robotClient, serviceName!, {
+  requestLogger: rcLogConditionally,
+});
+
+let canvasEl: HTMLCanvasElement;
+const initialCameras = filterSubtype($components, 'camera');
+let cameraName =
+  initialCameras.length > 0 ? initialCameras[0]!.name : undefined;
+let detections: Detection[] = [];
+let classifications: Classification[] = [];
+let ranOnce = false;
+
+// from https://stackoverflow.com/a/15666143
+let pixelRatio: number;
+{
+  const ctx = document.createElement('canvas').getContext('2d')!;
+  const dpr = window.devicePixelRatio || 1;
+  /* eslint-disable @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-member-access */
+  const bsr = ((ctx as any).webkitBackingStorePixelRatio ||
+    (ctx as any).mozBackingStorePixelRatio ||
+    (ctx as any).msBackingStorePixelRatio ||
+    (ctx as any).oBackingStorePixelRatio ||
+    (ctx as any).backingStorePixelRatio ||
```

Comment:
This property has been deprecated, and as far as I can tell from forum spelunking it's always been `1` on chrome, and used to be set in safari but is nowadays just `1`. Was this added to solve a specific problem or can we just simplify to just using `window.devicePixelRatio`?

Reply:
yep!